docs/hardhat-network/reference/README.md

```diff
@@ -0,0 +1,407 @@
+# Hardhat Network Reference
+
+## Supported hardforks
+
+- byzantium
+- constantinople
+- petersburg
+- istanbul
+- muirGlacier
+
+## Config
+
+- Default behavior: Hardhat will automatically run in-process when `defaultNetwork` is empty or set to `hardhat`.
+
+### Mining modes
+
+<!-- TODO: add a link to the explanation of mining modes -->
+
+You can configure the mining behavior under your Hardhat Network settings:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: 5000
+    }
+  }
+}
+```
+
+In this example, automining is disabled and interval mining is set so that a new block is generated every 5 seconds. You can also configure interval mining to generate a new block after a random delay:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: [3000, 6000]
+    }
+  }
+}
+```
+
+In this case, a new block will be mined after a random delay of between 3 and 6 seconds. For example, the first block could be mined after 4 seconds, the second block 5.5 seconds after that, and so on.
+
+#### Manual mining
+
+You can disable both mining modes like this:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: 0
+    }
+  }
+}
+```
+
+This means that no new blocks will be mined by the Hardhat Network, but you can manually mine new blocks using the `evm_mine` RPC method. This will generate a new block that will include as many pending transactions as possible.
+
+### Mempool behavior
+
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that:
+
+- Transactions with a higher gas price are included first
+- If two transactions can be included and both have the same gas price, the one that was received first is included first
+- If a transaction is invalid (for example, its nonce is lower than the nonce of the address that sent it), the transaction is dropped.
+
+You can get the list of pending transactions that will be included in the next block by using the "pending" block tag:
+
+```js
+const pendingBlock = await network.provider.send("eth_getBlockByNumber", [
+  "pending",
+  false,
+]);
+```
+
+### Removing and replacing transactions
+
+Transactions in the mempool can be removed using the `hardhat_dropTransaction` method:
+
+```js
+const txHash = "0xabc...";
+await network.provider.send("hardhat_dropTransaction", [txHash]);
+```
+
+You can also replace a transaction by sending a new one with the same nonce as the one that it's already in the mempool but with a higher gas price. Keep in mind that, like in Geth, for this to work the new gas price has to be at least 10% higher than the gas price of the current transaction.
```

Comment:
```suggestion
You can also replace a transaction by sending a new one with the same nonce as the one that it's already in the mempool but with a higher gas price. Keep in mind that, like in Geth, for this to work the new gas/fees prices has to be at least 10% higher than the gas price of the current transaction.
```

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/5dee7c7aceb2857be602261237b276c993f50dc9